platform/storage/zephyr/nvs/src/mender-storage.c

```diff
@@ -217,6 +219,67 @@ mender_storage_delete_deployment_data(void) {
 #ifdef CONFIG_MENDER_CLIENT_ADD_ON_CONFIGURE
 #ifdef CONFIG_MENDER_CLIENT_CONFIGURE_STORAGE
 
+mender_err_t
+mender_storage_save_update_state(mender_update_state_t state, const char *artifact_type) {
+    assert(artifact_type != NULL);
+
+    size_t artifact_type_len;
+
+    if (nvs_write(&mender_storage_nvs_handle, MENDER_STORAGE_NVS_UPDATE_STATE, state, sizeof(state)) != sizeof(state)) {
```

Comment:
> When a rewrite of the same data already stored is attempted, nothing is written to flash, thus 0 is returned.

Can this happen? That we call update with the same state that is already stored? If so, we should also accept `0` to be returned.

Reply:
Yup! Good catch!